test/corecel/data/HyperslabIndexer.test.cc

```diff
@@ -0,0 +1,112 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2020-2023 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file corecel/data/HyperslabIndexer.test.cc
+//---------------------------------------------------------------------------//
+
+#include "corecel/data/HyperslabIndexer.hh"
+
+#include "corecel/cont/Range.hh"
+
+#include "celeritas_test.hh"
+
+namespace celeritas
+{
+namespace test
+{
+//---------------------------------------------------------------------------//
+
+TEST(HyperslabIndexerTest, 2D)
+{
+    HyperslabIndexer<2> hi({3, 4});
+    HyperslabInverseIndexer<2> hii({3, 4});
```

Comment:
Do you think these would be more readable:
```suggestion
    HyperslabIndexer<2> to_index({3, 4});
    HyperslabInverseIndexer<2> from_index({3, 4});
```
(or `to_coords` or `index_to_coords`)?

Also you're referencing a temporary during this construction, which is an error: make sure to do
```c++
  Data const dims{3, 4};
  HyperslabIndexer<2> to_index(dims);
```